scripts/deploy.sh

```diff
@@ -4,6 +4,32 @@ set -e
 [[ "$(cast chain --rpc-url="$ETH_RPC_URL")" == "goerli" ]] || { echo "Please set a Goerli ETH_RPC_URL"; exit 1; }
 [[ "$ETHERSCAN_API_KEY" ]] || { echo "Please set ETHERSCAN_API_KEY"; exit 1; }
 
-make && \
-  dapp create DssSpell | \
-  xargs ./scripts/verify.py DssSpell
+SOURCE="src/test/config.sol"
+KEY_SPELL="deployed_spell"
+KEY_TIMESTAMP="deployed_spell_created"
+KEY_BLOCK="deployed_spell_block"
+
+make && spell_address=$(dapp create DssSpell)
+
+./scripts/verify.py DssSpell "$spell_address"
+
+# edit config.sol to add the deployed spell address
+sed -Ei "s/($KEY_SPELL: *address\()(0x[[:xdigit:]]{40}|0)\)/\1$spell_address)/" "$SOURCE"
+
+# get tx hash from contract address, created using an internal transaction
+TXHASH=$(curl "https://api-goerli.etherscan.io/api?module=account&action=txlistinternal&address=$spell_address&startblock=0&endblock=99999999&sort=asc&apikey=$ETHERSCAN_API_KEY" | jq -r ".result[0].hash")
+
+# get deployed contract timestamp and block number info
+timestamp=$(cast block "$(cast tx "${TXHASH}"|grep blockNumber|awk '{print $2}')"|grep timestamp|awk '{print $2}')
+block=$(cast tx "${TXHASH}"|grep blockNumber|awk '{print $2}')
+
+# edit config.sol to add the deployed spell timestamp and block number
+sed -i "s/\($KEY_TIMESTAMP *: *\)[0-9]\+/\1$timestamp/" "$SOURCE"
+sed -i "s/\($KEY_BLOCK *: *\)[0-9]\+/\1$block/" "$SOURCE"
+
+echo -e "Network: $(cast chain)"
+echo "config.sol updated with deployed spell: $spell_address, timestamp: $timestamp and block: $block"
+
+# commit edit change to config.sol
+(set -x; git add src/test/config.sol)
+(set -x; git commit -m "add deployed spell info")
```

Comment:
Added tests check in 8581a61

Reply:
As discussed in chat I'm testing some improvements to beef the script up with more checks before committing the changes as `make test` is a bit fragile an could return a zero status even if tests doesn't pass.

Reply:
Let me know if a56a15a and b68131f give more assurance, the extra check was inspired by the `dapp upgrade` one (see [here](https://github.com/dapphub/dapptools/blob/master/src/dapp/libexec/dapp/dapp-upgrade#L10))

Reply:
Discussing with @olivdb in chat, I've addressed the request of adding a way to stash and then reload local staging area in 2114dc1 (note that the stash would happen before the script makes edits and then it's reloaded before commiting in case the dev exit the commit edit prompt)

Reply:
after some testing and further discussions, using `trap` we ensure we always exit with restoring the staging area.